unblob/handlers/filesystem/yaffs/yaffs2.py

```diff
@@ -0,0 +1,172 @@
+import io
+from pathlib import Path
+from typing import Optional
+
+from structlog import get_logger
+
+from unblob.file_utils import File, read_until_past, snull
+from unblob.handlers.filesystem.yaffs.utils import (
+    C_DEFINITIONS,
+    YAFFS2Chunk,
+    YAFFS2Entry,
+    YAFFSConfig,
+    YAFFSFileVar,
+    YAFFSParser,
+    decode_file_size,
+    is_valid_header,
+    iterate_over_file,
+)
+from unblob.models import Extractor, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+
+class YAFFS2Parser(YAFFSParser):
+    def build_chunk(self, spare: bytes, config: YAFFSConfig) -> YAFFS2Chunk:
+        # images built without ECC have two superfluous bytes before the chunk ID.
+        if not config.ecc:
+            # adding two null bytes at the end only works if it's LE
+            spare = spare[2:] + b"\x00\x00"
+
+        yaffs2_packed_tags = self._struct_parser.parse(
+            "yaffs2_packed_tags_t", spare, self.config.endianness
+        )
+        logger.debug(
+            "yaffs2_packed_tags_t",
+            yaffs2_packed_tags=yaffs2_packed_tags,
+            config=config,
+            _verbosity=3,
+        )
+
+        return YAFFS2Chunk(
+            chunk_id=yaffs2_packed_tags.chunk_id,
+            seq_number=yaffs2_packed_tags.seq_number,
+            byte_count=yaffs2_packed_tags.byte_count,
+            object_id=yaffs2_packed_tags.object_id,
+        )
+
+    def parse(self):
```

Comment:
most of parser is the same for v1 and v2, except the entry creation

Reply:
Indeed. Generalized in `YAFFSParser` that rely on specialized functions such as `build_chunk` and `build_entry`.